### ISSUE TITLE
The labels of training after v8_transform is wrong,when visualize it ? 

### ISSUE THREAD

**Issue by SyGoing** [ultralytics/ultralytics#842]
### Search before asking

- [X] I have searched the YOLOv8 [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

when I train my own data with YOLO8, I found the labels of bboxes are shift with wrong way in the visualized batch data . ,the mosaic or the dataset transform got wrong.

**Comment by Laughing-q** [comment]
@SyGoing hi, please provide more info, such as ultralytics package version and your training command. Also please show us the visualized batch data if you don't mind. :)

**Comment by SyGoing** [comment]
@Laughing-q  
![image](https://user-images.githubusercontent.com/27557847/217145893-0c247f18-38d8-457b-a49c-386b65a9edbd.png)

**Comment by SyGoing** [comment]
def train_v8_detect():
    overrides = {"data": "custom_boat.yaml",
                 "resume":False,
                 "model":"yolov8n.pt",
                 "imgsz": 640,
                 "epochs": 300,
                 "save": True,
                 "device":0,
                 "batch":32,
                 "pretrained": True,
                 "v5loader":True
                }
    CFG.data = "custom_boat.yaml"

    # Trainer
    trainer = detect.DetectionTrainer(overrides=overrides)
    trainer.train()

    # Validator
    val = detect.DetectionValidator(args=CFG)
    val(model=trainer.best)  # validate best.pt

**Comment by SyGoing** [comment]
And,when I change the dataloader to v5loader ,the train labels are correct.

**Comment by SyGoing** [comment]
> @SyGoing hi, please provide more info, such as ultralytics package version and your training command. Also please show us the visualized batch data if you don't mind. :)

the version is yolov8.0.29

**Comment by Laughing-q** [comment]
@SyGoing can't reproduce your issue with `coco128.yaml`, I just replaced your data file to coco128.yaml but my visualized image looks good.
my script:
```python
from ultralytics.yolo.v8 import detect


def train_v8_detect():
    overrides = {
        "data": "coco128.yaml",
        "resume": False,
        "model": "yolov8n.pt",
        "imgsz": 640,
        "epochs": 300,
        "save": True,
        "device": 0,
        "batch": 32,
        "pretrained": True,
        "v5loader": False,
    }
    trainer = detect.DetectionTrainer(overrides=overrides)
    trainer.train()
if __name__ == "__main__":
    train_v8_detect()
```
![train_batch0](https://user-images.githubusercontent.com/61612323/217148498-8f9ea37f-edd9-42ca-9596-7df8505f4942.jpg)

**Comment by Laughing-q** [comment]
@SyGoing please check your dataset and try to delete the old .cache files to force re-cache.

**Comment by SyGoing** [comment]
Thanks for your reply,  when I delete the old cache and train it again, the labels change correct.  And, When I close it, train it again without delete the caches, the problem come again.

**Comment by Laughing-q** [comment]
@SyGoing I tested again and the existed cache files works fine to me. could you try coco128.yaml to see if this issue still happen?

**Comment by FreedomLiX** [comment]
@Laughing-q v8_transforms, when set the argument "stretch=True",  LetterBox() ，unused. 
so the img' shape do not convert to (640 640).   bug? @

**Comment by Laughing-q** [comment]
@FreedomLiX the image would be converted to 640x640 without any padding if stretch=True.
